Shell/LineEditor.cpp

```diff
@@ -2,10 +2,15 @@
 #include "GlobalState.h"
 #include <ctype.h>
 #include <stdio.h>
+#include <sys/ioctl.h>
 #include <unistd.h>
 
 LineEditor::LineEditor()
 {
+    struct winsize ws;
+    int rc = ioctl(STDOUT_FILENO, TIOCGWINSZ, &ws);
+    if (rc == 0)
```

Comment:
Do we expect this to fail with anything but `ENOTTY`? If not, maybe it should `perror()`/`ASSERT` this is not the case.

Reply:
I'm not 100% sure as I'm not very strong w.r.t signals/`ioctl`. I'll just add the `perror()` for now?

Reply:
No, getting `ENOTTY` is fine if stdout is not a TTY (and yeah, this should probably use stderr as well) — it's everything else that's unexpected. (But please verify this is the case by running `sh 2>/dev/null`)